content/blog/2024-06-27-using-bitnami-valkey-chart/index.md

```diff
@@ -0,0 +1,182 @@
+
+```bash
+$ kubectl run -it  python-redis --image=bitnami/python -- bash
+root@python-redis:/app# pip install redis
+Collecting redis
+Downloading redis-5.0.6-py3-none-any.whl.metadata (9.3 kB)
+Downloading redis-5.0.6-py3-none-any.whl (252 kB)
+  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 252.0/252.0 kB 942.1 kB/s eta 0:00:00
+Installing collected packages: redis
+Successfully installed redis-5.0.6
+```
+
+You are now ready to run the script.
+
+```bash
+root@python-redis:/app# python
+Python 3.12.4 (main, Jun  7 2024, 04:30:17) [GCC 12.2.0] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>> import redis
+>>> creds_provider = redis.UsernamePasswordCredentialProvider("", "test_pwd")
+>>> user_connection = redis.Redis(host="myvalkey-master.valkey", port=6379, credential_provider=creds_provider)
+>>> 
+>>> user_connection.ping()
+True
+>>> 
+>>> user_connection.set('foo_key', 'bar')
+True
+>>> 
+>>> user_connection.get('foo_key')
+b'bar'
+>>> 
+>>> user_connection.hset('user-session:123', mapping={
+...     'name': 'John',
+...     "surname": 'Smith',
+...     "company": 'Redis',
+...     "age": 29
+... })
+4
+>>> 
+>>> user_connection.hgetall('user-session:123')
+{b'name': b'John', b'surname': b'Smith', b'company': b'Redis', b'age': b'29'}
+```
+
+## Using Valkey as a cache for WordPress
+
+Valkey can also work as an object cache for a WordPress deployment. The following steps will show you how to set up this scenario.
+
+First, it is needed to create the configuration for the WordPress deployment, in this case, Valkey’s parameters and the user credential for the site administrator are set.
+
+```bash
+$ cat > settings.yaml <<EOF
+wordpressPassword: "wp_pwd"
+wordpressExtraConfigContent: |
+  define( 'WP_REDIS_HOST', 'myvalkey-master.valkey' );
+  define( 'WP_REDIS_PORT', 6379 );
+  define( 'WP_REDIS_PASSWORD', 'test_pwd' );
+  define( 'WP_REDIS_PREFIX', 'wp_redis' );
+EOF
+```
+
+Once you have the settings parameters, it is time to deploy WordPress in its namespace
+
+```bash
+$ kubectl create namespace wordpress
+$ helm install mywp bitnami/wordpress -f settings.yaml --namespace wordpress
+```
+
+Wait for the deployment to complete and check that the service has been assigned an external IP
+
+```bash
+$ kubectl get svc --namespace wordpress
+NAME                   TYPE         CLUSTER-IP     EXTERNAL-IP  PORT(S)         AGE
+service/mywp-mariadb   ClusterIP    10.106.11.191  <none>       3306/TCP                    74s
+service/mywp-wordpress LoadBalancer 10.100.136.181 192.168.49.50   80:31116/TCP,443:31119/TCP   74s
+```
+
+Using the external IP, browse to the admin page and log in using the previously set credentials.
+
+You will be able to see the WordPress administrator UI where you install and activate the Redis Object Cache plugin. After enabling it, wait for some metrics to appear
```

Comment:
You can look for it in the plugin administrator and install it from there. I mean, it is not a third party plugin that you need to download and install.
Not sure if you are asking for the link because of this.

Reply:
gotcha. No change needed if this is accessible without a link directly.